src/Tokenizer/Transformer/AmpersandTransformer.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Transformer;
+
+use PhpCsFixer\Tokenizer\AbstractTransformer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Transforms ampersand T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG and T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG into single CT::T_AMPERSAND.
```

Comment:
now i'm getting lost.
Idea of Transformers is that a single token, eg `array`, can serve different purposes depending on usage (like maybe it's declaring a new array ? maybe it's typehinting?). We don't know which is the case, we don't want to check context each time, so we have transformer to make the context explicit by new CustomTokens.

here, you do the opposite. you convert explicit tokens into singe token, loosing the knowledge about context. why is that?

Reply:
Mostly as the idea of T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG/T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG is not really helpful.

From what I've found these new 2 tokens were introduced to help with type intersection, and they do not.

If the token is T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG then we are sure it's not type intersection, but when type is T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG we still don't know - can by function reference or bitwise operator followed by constant. Also the [bug](https://bugs.php.net/bug.php?id=81342) (if it is bug) makes it more confusing.